redis-om-spring/src/main/java/com/redis/om/spring/search/stream/predicates/tag/EndsWithPredicate.java

```diff
@@ -0,0 +1,37 @@
+package com.redis.om.spring.search.stream.predicates.tag;
+
+import com.redis.om.spring.metamodel.SearchFieldAccessor;
+import com.redis.om.spring.repository.query.QueryUtils;
+import com.redis.om.spring.search.stream.predicates.BaseAbstractPredicate;
+import redis.clients.jedis.search.querybuilder.Node;
+import redis.clients.jedis.search.querybuilder.QueryBuilders;
+import redis.clients.jedis.search.querybuilder.QueryNode;
+
+public class EndsWithPredicate<E, T> extends BaseAbstractPredicate<E, T> {
+
+  private final T value;
+
+  public EndsWithPredicate(SearchFieldAccessor field, T value) {
+    super(field);
+    this.value = value;
+  }
+
+  public T getValue() {
+    return value;
+  }
+
+  @Override
+  public Node apply(Node root) {
```

Comment:
Seems like we can reuse the code from `StartWithPredicate` , maybe we can create a new base class ?

Reply:
Let do a separate ticket to clean up all the predicates and extract the templates to constants.